### ISSUE TITLE
Experimental DelayIdleTicker (zero cpu idle)

### ISSUE THREAD

**Issue by bencoman** [guillep/pharo-core#2871]
Moved from https://pharo.fogbugz.com/f/cases/22549/Experimental-DelayIdleTicker-zero-cpu-idle

My recent cleanup/refactoring of the delay scheduler hierarchy left me with an insight into a cool experiment to facilitate a zero CPU idle Image. I've implemented a quick proof-of-concept that I'll need help with critical analysis to identify corner cases and stress test it.  PR incoming soon.

**Comment by bencoman** [comment]
I had seen some action around releasing but missed the notice of an official freeze. 
This fixes older Priority 2 Case 22400 "Delays are not working properly" (hence matching priority)
but the existing PR[1] for that is simpler if you want to minimize changes. 
[1] https://github.com/pharo-project/pharo/pull/1873

**Comment by bencoman** [comment]
I should present this again.

**Comment by sbragagnolo** [comment]
I would enjoy to annoy joel with this message

**Comment by guillep** [comment]
Hi @bencoman , could you tell what would be some follow up actions in here?
I find the issue description a bit dry...

**Comment by bencoman** [comment]
This was prior to work I hear is being done on the Idle VM and probably has been superseded by that work. @estebanlm @tesonep ?

IIRC the idea I was working on was that it was "hard to understand" how (possibly)native-sleeping the Pharo application in Pharo's lowest priority "(10) Idle process" thread would interact with delay scheduling by the highest priority process "(80) DelayXXScheduler" and timely response there to TimingSemaphore signals.  I thought it would simply VM/native interaction for the (10)priority process to set a flag then signal TimingSemaphore for the (80)priority process to perform the native-sleep.  Then when VM woke from native-sleep the (80)priority process is already running to decide which Delay-waiting thread to wake up.  But I didn't get as far as actually making a callout to a native-sleep FFI function.

In summary, it was only half baked and likely now out of date, so probably best to just close the Issue.  The residual value of my ideas above may be worth momentary consideration by those working on the Idle VM.

**Comment by MarcusDenker** [comment]
", so probably best to just close the Issue."

 I will do that
